geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/FrequencyZ3.scala

```diff
@@ -0,0 +1,144 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import java.util.Date
+
+import com.clearspring.analytics.stream.frequency.{CountMinSketch, RichCountMinSketch}
+import com.vividsolutions.jts.geom.Geometry
+import org.joda.time.{DateTime, DateTimeZone, Seconds, Weeks}
+import org.locationtech.geomesa.curve.Z3SFC
+import org.opengis.feature.simple.SimpleFeature
+
+/**
+  * Estimates frequency counts at scale. Tracks geometry and date attributes as a single value.
+  *
+  * @param geomIndex geometry attribute index in the sft
+  * @param dtgIndex date attribute index in the sft
+  * @param precision number of bits of z-index that will be used
+  * @param eps (epsilon) with probability at least @see confidence, estimates will be within eps * N
+  * @param confidence percent - with probability at least confidence, estimates will be within @see eps * N
+  */
+class FrequencyZ3(val geomIndex: Int,
+                  val dtgIndex: Int,
+                  val precision: Int,
+                  val eps: Double = 0.005,
+                  val confidence: Double = 0.95) extends Stat {
+
+  override type S = FrequencyZ3
+
+  private val mask = Frequency.getMask(precision)
+
+  private [stats] val sketches = scala.collection.mutable.Map.empty[Short, CountMinSketch]
```

Comment:
One ```CountMinSketch``` per week?  That seems like a lot.  (Thinking aloud.)  GDELT has data from 1970 through today, so that's 46 years (rounding up), so that's about 52 * 46 = 2392 sketches for all of GDELT (through December 2016).  How much memory would that take?  Or am I misunderstanding how this is being stored?